sktime/forecasting/compose/_enbpi_forecaster.py

```diff
@@ -0,0 +1,219 @@
+        )
+        self.bootstrap_transformer = bootstrap_transformer
+        self.random_state = random_state
+
+        super().__init__()
+
+        from tsbootstrap import MovingBlockBootstrap
+
+        self.bootstrap_transformer_ = (
+            clone(bootstrap_transformer)
+            if bootstrap_transformer is not None
+            else MovingBlockBootstrap()
+        )
+
+    def _fit(self, X, y, fh=None):
+        self._fh = fh
+        self._y_ix_names = y.index.names
+
+        # random state handling passed into input estimators
+        self.random_state_ = check_random_state(self.random_state)
+
+        # fit/transform the transformer to obtain bootstrap samples
+        bs_ts_index = list(
+            self.bootstrap_transformer_.bootstrap(y, test_ratio=0, return_indices=True)
+        )
+        self.indexes = np.stack(list(map(lambda x: x[1], bs_ts_index)))
+        bootstapped_ts = list(map(lambda x: x[0], bs_ts_index))
+
+        self.forecasters = []
+        self.preds = []
+        # Fit Models per Bootstrap Sample
+        for bs_ts in bootstapped_ts:
+            bs_df = pd.DataFrame(bs_ts, index=y.index)
+            forecaster = clone(self.forecaster_)
+            forecaster.fit(y=bs_df, fh=fh, X=X)
+            self.forecasters.append(forecaster)
+            prediction = forecaster.predict(fh=y.index, X=X)
+            self.preds.append(prediction)
+
+        return self
+
+    def _predict(self, X, fh=None):
+        # Calculate Prediction Intervals using Bootstrap Samples
+
+        preds = [forecaster.predict(fh=fh, X=X) for forecaster in self.forecasters]
+
+        return pd.DataFrame(
+            np.stack(preds, axis=0).mean(axis=0),
+            index=list(fh.to_absolute(self.cutoff)),
+            columns=self._y.columns,
+        )
+
+    def _predict_interval(self, fh, X, coverage):
+        from fortuna.conformal import EnbPI
+
+        preds = []
+        for forecaster in self.forecasters:
+            preds.append(forecaster.predict(fh=fh, X=X))
+
+        train_targets = self._y.copy()
+        train_targets.index = pd.RangeIndex(len(train_targets))
+        intervals = []
+        for cov in coverage:
+            conformal_intervals = EnbPI().conformal_interval(
+                bootstrap_indices=self.indexes,
+                bootstrap_train_preds=np.stack(self.preds),
+                bootstrap_test_preds=np.stack(preds),
+                train_targets=train_targets,
+                error=1 - cov,
+            )
+            intervals.append(conformal_intervals)
+
+        cols = pd.MultiIndex.from_product(
+            [self._y.columns, coverage, ["lower", "upper"]]
+        )
+        fh_absolute = fh.to_absolute(self.cutoff)
+        fh_absolute_idx = fh_absolute.to_pandas()
+        pred_int = pd.DataFrame(
+            np.concatenate(intervals, axis=1), index=fh_absolute_idx, columns=cols
+        )
+        return pred_int
+
+    def _update(self, y, X=None, update_params=True):
```

Comment:
I think this is equivalent to the default, so could be removed?

Reply:
Done